packages/node/src/behaviors/operational-credentials/OperationalCredentialsServer.ts

```diff
@@ -313,22 +313,29 @@ export class OperationalCredentialsServer extends OperationalCredentialsBehavior
             );
         }
 
-        if (timedOp.forUpdateNoc) {
+        if (timedOp.forUpdateNoc === false) {
             throw new StatusResponseError(
-                `addNoc received after csr request was invoked for UpdateNOC.`,
+                `UpdateNoc received after csr request was invoked for UpdateNOC.`,
                 StatusCode.ConstraintError,
             );
         }
 
-        if (timedOp.hasRootCert) {
-            throw new StatusResponseError(
-                "Trusted root certificate added in this session which is now allowed for UpdateNOC.",
-                StatusCode.ConstraintError,
-            );
+        if (timedOp.rootCertSet) {
```

Comment:
I need a full test execution log please because this check should not be triggered at all because who should set an trusted root before? I needs to understand why this happens because in fact this should not be triggered